### PR TITLE
[feat]: [CDS-77038]: code pulse added

### PR DESCRIPTION
**Pull request by vinit1968**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52500)

### CODE REVIEW

File: 935-subscription-sdk/src/main/java/io/harness/subscription/helpers/StripeHelper.java [hunk 1]
```diff
@@ -22,6 +25,8 @@
 import java.util.List;
 import java.util.Optional;
 
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = false,
+        components = {HarnessModuleComponent.CDS_APPROVALS, HarnessModuleComponent.CDS_DASHBOARD})
 public interface StripeHelper {
```
Comment (Abhishek-CDC) [inline 935-subscription-sdk/src/main/java/io/harness/subscription/helpers/StripeHelper.java:30]: Please check if this file belong to CDS

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/artifacts/bamboo/BambooArtifactDelegateRequest.java [hunk 2]
```diff
@@ -32,7 +32,7 @@
 import lombok.EqualsAndHashCode;
 import lombok.Value;
 
-@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_APPROVALS})
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_ARTIFACTS,HarnessModuleComponent.CDS_APPROVALS})
 @Value
```
Comment (Abhishek-CDC) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/artifacts/bamboo/BambooArtifactDelegateRequest.java:36]: I think 'unitCoverageRequired' should be false and component should be only CDS_ARTIFACTS

File: 935-subscription-sdk/src/main/java/io/harness/subscription/helpers/StripeHelper.java (hunk 1 above)
Comment (Abhishek-CDC) [reply 935-subscription-sdk/src/main/java/io/harness/subscription/helpers/StripeHelper.java:30]: Also i think it is unrelated to dashboards

Comment (vinit1968) [reply 935-subscription-sdk/src/main/java/io/harness/subscription/helpers/StripeHelper.java:30]: I have made the changes.

**Comment by Abhishek-CDC** [comment]
trigger uts

File: 960-api-services/src/main/java/io/harness/artifacts/githubpackages/beans/SnapshotVersions.java [hunk 3]
```diff
@@ -14,7 +14,8 @@
 import javax.xml.bind.annotation.XmlRootElement;
 import lombok.Data;
 
-@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_ARTIFACTS})
+@CodePulse(
+    module = ProductModule.CDS, unitCoverageRequired = false, components = {HarnessModuleComponent.CDS_ARTIFACTS})
```
Comment (naman-goenka) [inline 960-api-services/src/main/java/io/harness/artifacts/githubpackages/beans/SnapshotVersions.java:18]: Can you please mark false here as well `960-api-services/src/main/java/io/harness/artifacts/githubpackages/beans/SnapshotVersion.java`

**Review by naman-goenka — COMMENTED** [review]
Added a comment

File: 960-api-services/src/main/java/io/harness/artifacts/githubpackages/beans/SnapshotVersions.java (hunk 3 above)
Comment (vinit1968) [reply 960-api-services/src/main/java/io/harness/artifacts/githubpackages/beans/SnapshotVersions.java:18]: It is already false.

**Comment by vinit1968** [comment]
trigger messagecheck

**Review by naman-goenka — APPROVED** [review]

**Review by Abhishek-CDC — APPROVED** [review]

**Review by puthrayaharness — APPROVED** [review]

**Comment by naman-goenka** [comment]
trigger review

**Review by abhinavmittal-harness — APPROVED** [review]
> Reviewed 18 of 18 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52500)* status: 19 file(s) to review. 2. unresolved discussions.

**Pull request merged**